src/Features/Core/Portable/SolutionCrawler/SolutionCrawlerProgressReporter.cs

```diff
@@ -19,109 +16,69 @@ internal partial class SolutionCrawlerRegistrationService : ISolutionCrawlerRegi
         /// </summary>
         private class SolutionCrawlerProgressReporter : ISolutionCrawlerProgressReporter
         {
-            private readonly IAsynchronousOperationListener _listener;
+            private int _progressStartCount;
+            private int _progressEvaluateCount;
 
-            // use event map and event queue so that we can guarantee snapshot and sequential ordering of events from
-            // multiple consumer from possibly multiple threads
-            private readonly SimpleTaskQueue _eventQueue;
-            private readonly EventMap _eventMap;
+            public SolutionCrawlerProgressReporter()
+            {
+                _progressStartCount = 0;
+                _progressEvaluateCount = 0;
```

Comment:
note: we have an actual Progress class somewhere in roslyn.  Can you use that instead?  It's already threadsafe and whatnot, so you don't have to implement the logic around that.   Look at IProgressTracker and its impls.

Reply:
I dont think that fits to this one.